### ISSUE TITLE
LocalPeopleResults in PnP Search doesn't show all 'members' from 'new' organisation following cross tenant synchronization work

### ISSUE THREAD

**Issue by JEBLuminate** [microsoft-search/pnp-modern-search#3851]
**Version used**
Ex: 4.6.1

**Describe the bug**
We use the PnP search in my organisation's intranet as an employee directory, so staff can look up colleagues to find out their contact information. We recently carried out cross tenant synchronization work to bring staff from a different part of the organisation that has its own SharePoint tenancy into our main tenancy as 'members' rather than guests. All of the members from this 'new' organisation appear in Microsoft Entra, but when we used the PnP Search only about 65 of the 300+ staff appear in the search results. It seems very random and there's no differences in the contact information for any of these 'new' members.

**To Reproduce**
Detailed steps to reproduce the behavior:
1. Type a name of a colleague from the 'new' organisation in the PnP search box
2. Search results come up 0.
3. Type the name of another colleague from the 'new' organisation in the PnP search box.
4. That colleague  appears in the search results. There are no differences between colleagues a and b from the new organisation, so both should appear.

**Expected behavior**
All staff from the 'member' organisation that have been moved into the new tenancy as part of cross tenant synchronization work should appear in the search results.

**Screenshots**
![Screenshot 2024-07-08 163041](https://github.com/microsoft-search/pnp-modern-search/assets/175034560/b739c164-741f-4e0d-9f0d-68782568b6c6)
53)

**Desktop (please complete the following information):**
Microsoft Edge

**Comment by kasperbolarsen** [comment]
LocalPeopleresults is based on the users in the User Profiles.  Can you
verify if the people missing are present in the UPA?

man. 8. jul. 2024 17.34 skrev JEBLuminate ***@***.***>:

**Comment by JEBLuminate** [comment]
Hello,
Yes. The missing people can be found in Microsoft Entra and are searchable across SharePoint.
See attached example of a colleague who can be found in the SharePoint search bar but not via the PnP search results.

![Screenshot 2024-07-08 165134](https://github.com/microsoft-search/pnp-modern-search/assets/175034560/77e0ee80-02a9-4368-a750-70fa39b5f84b)
![Screenshot 2024-07-08 165224](https://github.com/microsoft-search/pnp-modern-search/assets/175034560/c0171046-990d-449d-b542-600d1c410e9b)

**Comment by kasperbolarsen** [comment]
LocalPeopleResult is not pulled from Entra but from the SharePoint User Profile Application.  The sync from Entra to UPA must be correct,  https://learn.microsoft.com/en-us/sharepoint/user-profile-sync
Otherwise the accounts will not be in the UPA and thus not in LocalPeopleResults.
https://learn.microsoft.com/en-us/sharepoint/install/user-profile-service-overview

**Comment by JEBLuminate** [comment]
Thanks you.
The guidance on the links you sent suggest the sync to the SharePoint User Profile Application can't be configured. 
The sync appears quite random - 100 people out of around 300 from the 'new' organisation appear in the PnP Search - so I'm not sure how best to advise my IT department on what to check to ensure the sync from Entra to UPA is correct as it clearly works for some users.
Do you have any advise on the best approach?

**Comment by kasperbolarsen** [comment]
No, not really.  If your own IT doesn't have the skills required I would recommend you to get in touch with one of the many consultancies that have experience on the topic.  As I recall it you might also have to set up additional sync jobs for custom properties and profile pictures.
